### PR TITLE
Bk/assorted small issues

### PR DESCRIPTION
**Pull request by bkincart**

### CODE REVIEW

File: __tests__/client.test.js [hunk 1]
```diff
@@ -62,6 +62,11 @@ describe("generator-engage:client", () => {
       expect(json.dependencies["webpack-cli"]).toBeDefined();
     });
 
+    it("aliases react-dom as @hot-loader/react-dom in dev dependencies", () => {
+      json = readPackageJson();
+      expect(json.devDependencies["react-dom"]).toEqual("npm:@hot-loader/react-dom");
+    });
+
```
Comment (bkincart) [inline __tests__/client.test.js:69]: @dpickett current behavior has this failing, saying it's inserting it with the dependency of ~16.13 (not the new devDependency), and the test on line 38 also failing, saying it's not inserting it in dependencies at all. Was having trouble getting console.logs to show up, and debuggers aren't allowing the test suite to run.

File: __tests__/client.test.js [hunk 2]
```diff
@@ -62,6 +62,11 @@ describe("generator-engage:client", () => {
       expect(json.dependencies["webpack-cli"]).toBeDefined();
```
Comment (bkincart) [inline __tests__/client.test.js:62]: While i was in there I noticed this -- should it be checking devDependencies/have it there, or should the description just say "dependency"? This is conflicting

Comment (dpickett) [reply __tests__/client.test.js:62]: dependencies in react are confusing. Everything gets bundled into one large js file anyway, but webpack-cli is required in the client to build in production.

File: __tests__/client.test.js (hunk 1 above)
Comment (dpickett) [reply __tests__/client.test.js:69]: this is not the dev dependency. the `devDependency` is `@hot-loader/react-dom@YOUR_REACT_VERSION`

**Comment by dpickett** [comment]
Ok - I think I rigged up what you were seeking in terms of using the hot loader dom dependency. It was not aliasing in dev and I think that's what you were seeking so I added that. 

I'm not sure I understand the motive or intent behind the change of `start` to `prod` - can you walk me through that Monday @bkincart? Thanks!

File: __tests__/client.test.js (hunk 1 above)
Comment (bkincart) [reply __tests__/client.test.js:69]: Interesting, when I ran `yarn add react-dom@npm:@hot-loader/react-dom --dev` in my local version that was encountering the error, it added `"react-dom": "npm:@hot-loader/react-dom"` verbatim into my devDependencies. Either way, your fix seems to be working so thank you for that :)

**Comment by bkincart** [comment]
This looks great, I tested locally and am going to go ahead and merge. Yes, I added a reminder to myself to touch base with you on that on Monday @dpickett ! :)

**Pull request merged**